hasher-matcher-actioner/hmalib/lambdas/api/content.py

```diff
@@ -69,6 +118,60 @@ def content() -> t.Optional[ContentObject]:
             return ContentObject.get_from_content_id(dynamodb_table, content_id)
         return None
 
+    @content_api.get("/pipeline-progress/", apply=[jsoninator])
+    def pipeline_progress() -> ContentPipelineProgress:
+        """
+        Build a history of the stages that this piece of content has gone
+        through and what their results were. Do not call this from anything but
+        a UI. This is not optimized for performance.
+        """
+        content_id = bottle.request.query.content_id or None
+        content_object = ContentObject.get_from_content_id(dynamodb_table, content_id)
+        preview_url = content_object.content_ref
+
+        # The result object will be gradually built up as records are retrieved.
+        result = ContentPipelineProgress(
+            content_id=content_id,
+            content_type=content_object.content_type,
+            content_preview_url=preview_url,
+            submitted_at=content_object.updated_at,
+            submission_additional_fields=list(content_object.additional_fields),
+        )
+
+        hash_records = PipelineHashRecord.get_from_content_id(
+            dynamodb_table, content_id
+        )
+        if len(hash_records) != 0:
+            result.hashed_at = min(hash_records, key=lambda r: r.updated_at).updated_at
```

Comment:
One of the goals for the pipeline progress indicator was to show the low latency. It actually surprised me too. :) 

Which is why the `min` made sense to me. Also, when you say "most recently..." you mean with the max() value, right? With the current 'min', it would be "earliest ...".